alphabase/io/tempmmap.py

```diff
@@ -104,12 +121,23 @@ def create_empty_mmap(shape: tuple, dtype: np.dtype, path: str = None, overwrite
     overwrite : bool , optional
         If True the file will be overwritten if it already exists.
         Defaults to False.
+    tmp_dir_name : str, optional
+        If specified the memory mapped array will be created in this directory.
 
     Returns
     -------
     str
         path to the newly created file.
     """
+    # redefine the temporary directory if a new location is given otherwise read from global variable
+    # this allows you to ensure that the correct temp directory location is used when working with multiple threads
+    if tmp_dir_name is not None:
+        if os.path.isdir(tmp_dir_name):  # ensure that directory exists
+            TEMP_DIR_NAME = tmp_dir_name
+        else:
+            raise ValueError(
+                "The directory in which the file should be created does not exist."
+            )
```

Comment:
Would I need to return the variable name as otherwise TEMP_DIR_NAME is only updated within the scope of the function and outside of the function we are still using the previously defined global variable?

Reply:
I wrote a function which returns the path name if it fulfills the conditions and otherwise raises a warning. This should let us keep the same functionality without having to use global variables.